homeassistant/components/unifiprotect/switch.py

```diff
@@ -350,42 +360,90 @@ def __init__(
+
+
+class ProtectPrivacyModeSwitch(RestoreEntity, ProtectSwitch):
+    """A UniFi Protect Switch."""
+
+    device: Camera
+
+    def __init__(
+        self,
+        data: ProtectData,
+        device: ProtectAdoptableDeviceModel,
+        description: ProtectSwitchEntityDescription,
+    ) -> None:
+        """Initialize an UniFi Protect Switch."""
+        super().__init__(data, device, description)
+
+        if self.device.is_privacy_on:
+            extra_state = self.extra_state_attributes or {}
+            self._previous_mic_level = extra_state.get(ATTR_PREV_MIC, 100)
+            self._previous_record_mode = extra_state.get(
+                ATTR_PREV_RECORD, RecordingMode.ALWAYS
+            )
+        else:
             self._previous_mic_level = self.device.mic_volume
             self._previous_record_mode = self.device.recording_settings.mode
-            await self.device.set_privacy(True, 0, RecordingMode.NEVER)
+
+    @callback
+    def _update_previous_attr(self) -> None:
+        if self.is_on:
+            self._attr_extra_state_attributes = {
+                ATTR_PREV_MIC: self._previous_mic_level,
+                ATTR_PREV_RECORD: self._previous_record_mode,
+            }
         else:
-            await self.entity_description.ufp_set(self.device, True)
+            self._attr_extra_state_attributes = {}
+
+    @callback
+    def _async_update_device_from_protect(self, device: ProtectModelWithId) -> None:
+        super()._async_update_device_from_protect(device)
+
+        # do not add extra state attribute on initialize
+        if self.entity_id:
+            self._update_previous_attr()
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the device on."""
+
+        self._previous_mic_level = self.device.mic_volume
+        self._previous_record_mode = self.device.recording_settings.mode
+        await self.device.set_privacy(True, 0, RecordingMode.NEVER)
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the device off."""
 
-        if self._switch_type == _KEY_PRIVACY_MODE:
-            assert isinstance(self.device, Camera)
-            _LOGGER.debug(
-                "Setting Privacy Mode to false for %s", self.device.display_name
-            )
-            await self.device.set_privacy(
-                False, self._previous_mic_level, self._previous_record_mode
-            )
-        else:
-            await self.entity_description.ufp_set(self.device, False)
+        extra_state = self.extra_state_attributes or {}
+        prev_mic = extra_state.get(ATTR_PREV_MIC, self._previous_mic_level)
+        prev_record = extra_state.get(ATTR_PREV_RECORD, self._previous_record_mode)
+        await self.device.set_privacy(False, prev_mic, prev_record)
+
+    async def async_added_to_hass(self) -> None:
+        """Restore extra state attributes on startp up."""
+        await super().async_added_to_hass()
+
```

Comment:
Would it make sense to use `extra_restore_state_data` here?

Reply:
On second thought, probably not since users probably find the attr values useful 😆